### PR TITLE
more precise $argc, $argv global var type

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/vimeo/psalm/pull/8473

### CODE REVIEW

File: tests/PHPStan/Analyser/data/cli-globals.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+function g($argc, $argv) {
+	\PHPStan\Testing\assertType('mixed', $argc);
+	\PHPStan\Testing\assertType('mixed', $argv);
+}
+
+function h() {
+	global $argc, $argv;
+	\PHPStan\Testing\assertType('mixed', $argc); // should be int<1, max>
+	\PHPStan\Testing\assertType('mixed', $argv); // should be non-empty-array<int, string>
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/cli-globals.php]: tried several things, but in the end wasn't able to identify this variable-scope.

therefore left it `mixed` for now.. I don't think thats a very common thing anyway
(and we can't be 100% sure whether `global $argc, $argv` actually references the builtin globals or a custom variable)

File: tests/PHPStan/Analyser/data/cli-globals.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace CliGlobals;
+
+\PHPStan\Testing\assertType('int<1, max>', $argc);
+\PHPStan\Testing\assertType('non-empty-array<int, string>', $argv);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/cli-globals.php]: $argc/$argv always count/contain the program name you pass `php test.php`, so actually you never have a empty type here.

actually this variable would only exist for a script run via the CLI SAPI, but I don't think we can differentiate CLI and WEB sapi.

FWIW, psalm also allows this 2 vars: https://psalm.dev/r/942a865551 and they are in the process of making them more precise in https://github.com/vimeo/psalm/pull/8473/files. I will look into making the super-globals more precise in a separate PR.

File: tests/PHPStan/Analyser/data/cli-globals.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+function f() {
+	\PHPStan\Testing\assertType('*ERROR*', $argc);
```
Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/data/cli-globals.php]: Please put this function in a use statement.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Please also test if these variables are locally created by the user and contain a different type.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**